projects/fluid_simulation/shaders.h

```diff
@@ -89,7 +89,7 @@ class Texture
     }
 
 private:
-    ProjApp* GetApp() const;
+    ppx::Application* GetApp() const;
```

Comment:
Is it necessary for the FLuidSim to know about app?

Also, you might be aware already, there's static method `Application::Get()` that returns the current app instance. You can use this to get both the Application and Device.

Reply:
No it is not, but that requires more refactoring. I wanted to make it in incremental steps to facilitate reviews.